util/noeintr.h

```diff
@@ -11,4 +11,11 @@
  */
 ssize_t noeintr_write(int, const void *, size_t);
 
+/**
+ * noeintr_close(fd):
+ * Close the file descriptor ${fd} per the close(2) system call, but handle
+ * EINTR appropriately.
```

Comment:
s/handle EINTR appropriately/retry on EINTR if the descriptor was not closed/

Also we should add "Unlike close(2), this function is not async-signal-safe."